site/cds_rdm/legacy/redirector.py

```diff
@@ -59,19 +63,66 @@ def legacy_files_redirect(legacy_id, filename):
     return redirect(url_path)
 
 
+def legacy_collection_redirect(collection_name):
+    """Redirection for legacy collections."""
+    cds_community_uuid = current_app.config.get(
+        "CDS_REDIRECTION_COLLECTIONS_MAPPING", {}
+    ).get(collection_name, None)
+    if not cds_community_uuid:
+        raise NoResultFound
+    community_data = get_community_by_uuid(cds_community_uuid)
+    query_params = request.args.copy()
+    query_params["q"] = query_params.pop("p", None)
+    if query_params["q"]:
+        url_path = url_for(
+            "invenio_app_rdm_communities.communities_detail",
+            pid_value=community_data.slug,
+            **query_params,
+        )
+    else:
+        url_path = url_for(
+            "invenio_app_rdm_communities.communities_home",
+            pid_value=community_data.slug,
+            **query_params,
+        )
+    return redirect(url_path)
+
+
+def legacy_search_redirect():
+    """Redirection for legacy search."""
```

Comment:
Can you be a bit more verbose with the docstring? It would be nice to read what is handling, and with one or more examples. E.g.
```
c=<colleciton name> -> /search?...
```